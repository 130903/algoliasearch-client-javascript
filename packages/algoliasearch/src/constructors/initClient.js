/* eslint import/namespace: [2, { allowComputed: true }] */
// @flow

import * as clientMethods from '../methods/client';
import attachParameters from './attachParameters';
import { createRequester } from '../request';

import type { ClientMethods, AppId, ApiKey } from '../types';

export default function initClient({
  appId,
  apiKey,
}: {
  appId: AppId,
  apiKey: ApiKey,
}): ClientMethods {
  if (appId === undefined) {
    throw new Error(`An appId is required. ${appId} was not valid.`);
  }
  if (apiKey === undefined) {
    throw new Error(`An apiKey is required. ${apiKey} was not valid.`);
  }

  const requester = createRequester(appId, apiKey);
  return attachParameters(clientMethods, requester);
}
