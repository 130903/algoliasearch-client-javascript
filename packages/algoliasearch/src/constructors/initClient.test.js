// @flow
import initClient from './initClient.js';

const validClientParams = {
  appId: 'some_app',
  apiKey: 'some_key',
};

it('initClient throws when it has too little parameters', () => {
  // $FlowFixMe --> type disallows this
  expect(() => initClient({})).toThrow();
  // $FlowFixMe --> type disallows this
  expect(() => initClient({ appId: '' })).toThrowErrorMatchingSnapshot();
  // $FlowFixMe --> type disallows this
  expect(() => initClient({ apiKey: '' })).toThrowErrorMatchingSnapshot();

  expect(() => initClient(validClientParams)).not.toThrow();
});

it('initClient contains the correct methods', () => {
  const client = initClient(validClientParams);
  expect(Object.keys(client)).toMatchSnapshot();
});
