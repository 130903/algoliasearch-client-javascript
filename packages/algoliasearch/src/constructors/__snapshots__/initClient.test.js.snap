// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client.search() works üê£ 1`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "indexName": "some_index",
        "query": "hello world",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/queries",
  "requestOptions": undefined,
  "requestType": "read",
}
`;

exports[`initClient contains the correct methods 1`] = `
Array [
  "addApiKey",
  "batch",
  "clearIndex",
  "copyIndex",
  "deleteApiKey",
  "deleteIndex",
  "getApiKey",
  "getLogs",
  "initIndex",
  "listApiKeys",
  "listIndexes",
  "moveIndex",
  "search",
  "updateApiKey",
  "requester",
]
`;

exports[`initClient throws when it has too little parameters 1`] = `
"An apiKey is required. undefined was not valid.

initIndex({\\"appId\\":\\"\\"})"
`;

exports[`initClient throws when it has too little parameters 2`] = `
"An appId is required. undefined was not valid.

initIndex({\\"apiKey\\":\\"\\"})"
`;
